Bearded.Utilities.Tests/Core/Random/SharedTests.cs

```diff
@@ -0,0 +1,441 @@
+            }
+        }
+        
+        public abstract class TheDoubleWithOneBoundMethod : BaseTests.RandomMethodWithOneParameter<double>
+        {
+            [Property]
+            public void ShouldReturnValueInRange(int seed, NormalFloat bound)
+            {
+                var b = bound.Get;
+                var (minimumValue, maximumValue) = b > 0
+                    ? (0d, b)
+                    : (b, 0d);
+                
+                ResultOfCallingWith(seed, b)
+                    .Should().BeInRange(minimumValue, maximumValue);
+            }
+
+            [Property]
+            public void ShouldReturnZeroForZeroBound(int seed)
+            {
+                ResultOfCallingWith(seed, 0).Should().Be(0);
+            }
+        }
+
+        public abstract class TheDoubleWithTwoBoundsMethod : BaseTests.RandomMethodWithTwoParameters<double>
+        {
+            [Property]
+            public void ShouldReturnValueInRange(int seed, NormalFloat bound1, NormalFloat bound2)
+            {
+                var (b1, b2) = (bound1.Get, bound2.Get);
+                var (minimumValue, maximumValue) = b1 < b2
+                    ? (b1, b2)
+                    : (b2, b1);
+
+                ResultOfCallingWith(seed, b1, b2)
+                    .Should().BeInRange(minimumValue, maximumValue);
+            }
+
+            [Property]
+            public void ShouldReturnBoundIfBoundsAreEqual(int seed, NormalFloat bound)
+            {
+                var b = (float) bound.Get;
+                ResultOfCallingWith(seed, b, b).Should().Be(b);
+            }
+        }
+
+        public abstract class TheSignMethod : BaseTests.RandomMethodWithoutParameters<int>
+        {
+            [Property]
+            public void ReturnsPlusOrMinusOne(int seed)
+            {
+                ResultOfCallingWith(seed).Should().BeOneOf(-1, 1);
+            }
+
+            [Property]
+            public void ReturnsBothValidValues(int seed)
+            {
+                Seed(seed);
+                sequence(100, CallMethod)
+                    .Should().Contain(new []{-1, 1});
+            }
+        }
+
+        public abstract class TheBoolMethod : BaseTests.RandomMethodWithoutParameters<bool>
+        {
+            [Property]
+            public void ReturnsBothPossibleValues(int seed)
+            {
+                Seed(seed);
+                sequence(100, CallMethod)
+                    .Should().Contain(new []{true, false});
+            }
+        }
+        
+        public abstract class TheBoolWithParameterMethod : BaseTests.RandomMethodWithOneParameter<bool, double>
+        {
+            [Property]
+            public void ReturnsOnlyFalseForZeroParameter(int seed)
+            {
+                ResultOfCallingWith(seed, 0).Should().Be(false);
+            }
+            
+            [Property]
+            public void ReturnsOnlyTrueForOneParameter(int seed)
```

Comment:
This reads as if we are giving the function one parameter, rather than the parameter `1`.

Reply:
I think it's clear in context. If you insist, suggestions as welcome, but then other methods should probably be renamed too for consistency.

Reply:
I thought the idea of this complicated test structure was so that the tested assertions were clear from the name of the test, which seems to clash with a reasoning that it's clear from context.

Up to you if you think there's anything clearer (you know I dislike the whole sentence-naming anyway), but the alternatives that come to mind are:

* ReturnsOnlyTrueForOne
* ReturnsOnlyTrueForParameterOne
* ReturnsOnlyTrueWhenCalledWithOne
* ReturnsOnlyTrueWhenProbabilityIsOne
* ReturnsOnlyTrueForProbabilityOne

Reply:
If you include the context, i.e. the class name, in the test name, I think it's quite clear.

In either case, there's a lot of different options, but unless we agree to some format I'd just leave it this way here for consistency within the same file.

Reply:
Even in a full sentence I don't see how the "one" isn't interpreted as a cardinal number in this case. Maybe it's a gold dress, blue dress thing where you can only see one or the other, but I can only find the name of this test misleading. I think just removing the "Parameter" bit takes away both of the bloat of the test name, and removes this confusion, but I won't insist further, since I'll be unlikely to work with these tests myself in the future.